File: leetcode/Offer06.cpp
#include <vector>
#include <algorithm>

using namespace std;

// Definition for singly-linked list.
struct ListNode {
  int val;
  ListNode *next;
  ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
  vector<int> reversePrint(ListNode *head) {
    vector<int> rst;
    while (head != NULL) {
      rst.emplace_back(head->val);
      head = head->next;
    }
    reverse(rst.begin(), rst.end());
    return rst;
  }
};

File: leetcode/Offer06.py
from typing import List


# Definition for singly-linked list.
class ListNode:

    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:

    def reversePrint(self, head: ListNode) -> List[int]:
        rst = []
        while head != None:
            rst.append(head.val)
            head = head.next
        # rst.reverse()
        # return rst
        return rst[::-1]